### PR TITLE
Switching findOneAndUpdate with findByIdAndUpdate

### PR DESCRIPTION
**Pull request by jim-sheldon**
And adding logging on errors, and database entries for download formats and queries.

### CODE REVIEW

File: verification/curator-service/api/src/controllers/cases.ts [hunk 1]
```diff
@@ -136,11 +167,20 @@ export default class CasesController {
...
-                {
-                    _id: user._id,
-                },
-                { $push: { downloads: { timestamp: new Date() } } },
+            await User.findByIdAndUpdate(
+                user._id,
+                { $push: { downloads: {
+                    timestamp: new Date(),
+                    format: req.body.format,
+                    query: req.body.query,
```
Comment (jim-sheldon) [inline verification/curator-service/api/src/controllers/cases.ts:175]: Should I leave `format` and `query` out here? Or make them `csv` and `{}`?

Comment (iamleeg) [reply verification/curator-service/api/src/controllers/cases.ts:175]: I would say pass on what's in the req, but check in swagger if they're required here and supply defaults if not.

**Review by iamleeg — APPROVED** [review]
assuming it passes tests now, let's get it out there and try it

File: verification/curator-service/api/src/controllers/cases.ts (hunk 1 above)
Comment (jim-sheldon) [reply verification/curator-service/api/src/controllers/cases.ts]: They're optional fields for backwards compatibility. Omitting them here.

**Pull request merged**